interpreter/exec/i32.ml

```diff
@@ -4,4 +4,11 @@ include Int.Make
   (struct
     include Int32
     let bitwidth = 32
+
+    let avgr_u x y =
+      let open Int64 in
+      let mask = of_int (-1) in
+      let x64 = logand mask (of_int32 x) in
+      let y64 = logand mask (of_int32 y) in
+      to_int32 (div (add (add x64 y64) one) (of_int 2))
```

Comment:
Shouldn't this be implemented generically in the Int functor instead of repeating it for each type?

Reply:
I thought of that initially but had trouble, since we extend i16 to int32, and i32 to i64.
But we can use i64 for all cases. The additional requirement then is for Rep to provide {to,of}_int64, in order to extend to Int64.t and narrow back down into Rep.t. I added this to RepType, and implemented them in the obvious way in i16.ml, i32.ml, and i64.ml. Let me know if that looks good.

(I thought of int, but I think it will be problematic on 32-bit platforms, since not all values of int32_t will be representable if we convert to int, I think we will have some problems)